salt/modules/swarm.py

```diff
@@ -0,0 +1,440 @@
+# -*- coding: utf-8 -*-
+"""
+Docker Swarm Module using Docker's Python SDK
+=============================================
+
+:codeauthor: Tyler Jones <jonestyler806@gmail.com>
+
+.. versionadded:: Oxygen
+
+The Docker Swarm Module is used to manage and create Docker Swarms.
+
+Dependencies
+------------
+
+- Docker installed on the host
+- Docker python sdk >= 2.5.1
+
+Docker Python SDK
+-----------------
+pip install -U docker
+
+More information: https://docker-py.readthedocs.io/en/stable/
+"""
+# Import python libraries
+from __future__ import absolute_import
+import salt.config
+import salt.loader
+import json
+
+try:
+    import docker
+    HAS_DOCKER = True
+except ImportError:
+    HAS_DOCKER = False
+
+__virtualname__ = 'docker'
+
+
+def __virtual__():
+    '''
+    Load this module if the docker python module is installed
+    '''
+    if HAS_DOCKER:
+        return __virtualname__
+    return False, 'The Docker module failed to load: Docker python module is not avaialble.'
```

Comment:
should probably say `The swarm`

Alternatively, a lot of this stuff could just be added to the `dockermod.py` file instead of being in a separate module.